### PR TITLE
Updated numba_utils.py for general_roc function

### PR DESCRIPTION
**Pull request by naman0210**

### CODE REVIEW

**Comment by naman0210** [comment]
from pysaliency.roc import general_roc
import hypothesis.strategies as st
from hypothesis import given, settings, assume, Verbosity

@settings(deadline=None)
@given(st.lists(st.floats(allow_infinity=False,allow_nan=False),min_size=1), st.lists(st.floats(allow_infinity=False,allow_nan=False),min_size=1))

def test_hypothesis(positives,negatives):
    positives = np.array(positives)
    negatives = np.array(negatives)
    assume (len(positives) == len(negatives))
    
    numba_output = general_roc_numba(positives,negatives)
    cython_output = general_roc(positives,negatives)
    assert np.isclose(numba_output[0],cython_output[0])
    assert (numba_output[1] == cython_output[1]).all()
    assert (numba_output[2] == cython_output[2]).all()

**Comment by matthias-k** [comment]
@naman0210 can you please add an actual test for the comparison to the cython implementation? I would like to have this in the repository as long as we keep both implementations. Also, could you please rebase the PR on top of the dev branch?

**Comment by matthias-k** [comment]
`tests/test_numba_utils.py` will be the right place, that file also already contains a hypothesis test.

File: pysaliency/numba_utils.py [hunk 1]
```diff
@@ -47,3 +47,43 @@ def _auc_for_one_positive(positive, negatives):
...
+    for i in range(len(all_values)):
+        theta = all_values[i]
+        while true_positive_count < positive_count and sorted_positives[true_positive_count] >= theta:
+            true_positive_count += 1
+        while false_positive_count < negative_count and sorted_negatives[false_positive_count] >= theta:
+            false_positive_count += 1
+        false_positive_rates[i+1] = float(false_positive_count) / negative_count
+        hit_rates[i+1] = float(true_positive_count) / positive_count
+
+    auc = np.trapz(hit_rates, false_positive_rates)
```
Comment (matthias-k) [inline pysaliency/numba_utils.py:88]: does this numpy call work with `nopython=True`? Anyway, I would suggest to call it in the `general_roc_numba` function and have `_general_roc_numba` only compute the hit rates and false positive rate

Comment (naman0210) [reply pysaliency/numba_utils.py:88]: yes this numpy call does seem to work. But we need the hit_rates and false_positive_rates to calculate the auc and those are being calculated in the  _general_roc_numba.

File: pysaliency/numba_utils.py [hunk 2]
```diff
@@ -47,3 +47,47 @@ def _auc_for_one_positive(positive, negatives):
...
+    else:
+        min_val = min(sorted_positives[len(positives)-1], sorted_negatives[len(negatives)-1])
+        max_val = max(sorted_positives[0], sorted_negatives[0]) + 1
+        all_values = np.hstack((max_val, positives, min_val))
+        all_values = np.sort(all_values)[::-1]
+
+    false_positive_rates = np.zeros(len(all_values) + 1)
+    hit_rates = np.zeros(len(all_values) + 1)
+    positive_count = len(positives)
+    negative_count = len(negatives)
```
Comment (matthias-k) [inline pysaliency/numba_utils.py:68]: these variables are never used

**Pull request merged**